src/org/opensolaris/opengrok/analysis/plain/AbstractSourceCodeAnalyzer.java

```diff
@@ -76,6 +77,8 @@ static protected void writeXref(JFlexXref lxref, Reader in, Writer out, Definiti
             lxref.reInit(in);
             lxref.annotation = annotation;
             lxref.project = project;
+            lxref.setScopesEnabled(RuntimeEnvironment.getInstance().isScopesEnabled());
```

Comment:
As far as I understand this and `AnalyzerGuru.java#writeXref()` are used only for testing. Otherwise during normal indexing the xref is created via this path:

```
IndexDatabase#addFile()
  FileAnalyzer fa = AnalyzerGuru.getAnalyzer() // fa can be e.g. CAnalyzer
  AnalyzerGuru#populateDocument()
    fa.analyze()
    AbstractSourceCodeAnalyzer#analyze()
      super.analyze()
        PlainAnalyzer#analyze()
          writeXref()
            xref = newXref() // xref can be e.g. CXref
            JFlexXref.write()

```

where the scopes/folding properties are in effect thanks to the inheritance and these lines in `IndexDatabase#addFile()`:

```
662          fa.setScopesEnabled(RuntimeEnvironment.getInstance().isScopesEnabled());
663          fa.setFoldingEnabled(RuntimeEnvironment.getInstance().isFoldingEnabled());
```

So having these properties set into the `writeXref()` seems like short-circuiting however there is no other way given the method is static.

The only nit I have is that the call to `reInit()` in `JFlexXref` above contains:

```
        scopes = new Scopes();
```

which is not necessary when scopes are disabled. Granted, this method does not do anything however it's a bit of wasted memory.